lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx

```diff
@@ -0,0 +1,427 @@
+
+    expect(onSearch).not.toHaveBeenCalled();
+    await userEvent.click(retryBtn);
+    expect(onSearch).toHaveBeenCalled();
+  });
+
+  it('fires onChange when the button with role option is clicked', async function() {
+    const onChange = jest.fn(),
+        { getByRole, getAllByRole } = rtlRender({
+          matches: [{ id: '1', displayName: 'Foo' }, { id: '2', displayName: 'Boo' }], onChange }),
+        inputElement = getByRole('combobox');
+
+    await userEvent.type(inputElement, 'o');
+    const optionBtns = getAllByRole('option');
+    await userEvent.click(optionBtns[1]);
+    expect(onChange).toHaveBeenCalledWith('Boo');
+  });
+
+  it('renders empty message if there are no results', function() {
+    const { container } = rtlRender(),
+        emptyMessage = container.querySelector('.nx-combobox__empty-message');
+
+    expect(emptyMessage).toBeTruthy();
+    expect(emptyMessage).toHaveTextContent('No Results Found');
+  });
+
+  it('sets the empty message from the `emptyMessage` prop', function() {
+    const { container } = rtlRender({ emptyMessage: 'asdfasdf' }),
+        emptyMessage = container.querySelector('.nx-combobox__empty-message');
+
+    expect(emptyMessage).toHaveTextContent('asdfasdf');
+  });
+
+  it('calls onSearch with the current trimmed search text if focus enters the component from elsewhere on the page ' +
+  'while there is an error', function() {
+    const onSearch = jest.fn(),
+        { getByRole} = rtlRender({ value: 'f', loadError: 'err', onSearch }),
+        inputElement = getByRole('combobox'),
+        anotherElement = document.createElement('button');
+
+    document.body.append(anotherElement);
+    anotherElement.focus();
+
+    expect(onSearch).not.toHaveBeenCalled();
+    inputElement.focus();
+    expect(onSearch).toHaveBeenCalledWith('f');
+  });
+
+  it('does not call onSearch if focus moves within the component while there is an error', function() {
+    const onSearch = jest.fn(),
+        { getByRole } = rtlRender({ value: 'f', loadError: 'err', onSearch }),
+        inputElement = getByRole('combobox');
+
+    expect(onSearch).not.toHaveBeenCalled();
+    inputElement.focus();
+    expect(onSearch).not.toHaveBeenCalled();
+
+    const retryBtn = getByRole('button', {name: /retry/i });
+    retryBtn.focus();
+
+    expect(onSearch).not.toHaveBeenCalled();
+    inputElement.focus();
+    expect(onSearch).not.toHaveBeenCalled();
+  });
+
+  it('does not call onSearch if focus moves into the component from an outside window while there is an error',
+      function() {
+        const onSearch = jest.fn(),
+            { getByRole } = rtlRender({ value: 'f', loadError: 'err', onSearch }),
+            inputElement = getByRole('combobox');
+
+        expect(onSearch).not.toHaveBeenCalled();
+        inputElement.focus();
+        expect(onSearch).not.toHaveBeenCalled();
+        (document.activeElement as HTMLElement).blur();
+        expect(onSearch).not.toHaveBeenCalled();
```

Comment:
Checking that it was never called by the end of the test is adequate
```suggestion
        inputElement.focus();
        (document.activeElement as HTMLElement).blur();
```